### DISCUSSION TITLE
How to add/remove relation from field.

### DISCUSSION THREAD

**Discussion by AaronHessler** [pocketbase/pocketbase#2394] [Q&A] [answered]
I have a collection of posts and I would like to keep track who liked the posts.
So I have a relation field called "likes" and I would like to add/remove a user with JavaScript if they liked (or didn't) the post.
So if they click on the like button it will update the relation field "likes" of the post and add their user as a relation. The same if they "unlike" the post.
Something around the lines of
`const record = await pb.collection('posts').update('POST_ID', "likes").addRelation( "USER_RECORD_ID");`
How would I do that without creating security risks?

**Comment by ganigeorgiev** [comment]
I don't understand the question.

Please provide more details about your collections structure (_you can export your collections as json from the Admin UI > Settings > Export collections_) and the expected outcome.

In general, access control in PocketBase is done via the collection API rules - https://pocketbase.io/docs/api-rules-and-filters/.

**Reply by AaronHessler** [reply]
I've improved the description of my problem a bit. Hope it helps.

**Reply by ganigeorgiev** [reply, marked as answer]
If you want a single user to be able to change only their "like" status without interfering with the others, it will be better to create a junction collection, eg. `postLikes` with 3 fields:

- `post` (single `relation` to the `posts` collection; you can also enable the "Cascade delete" option)
- `user` (single `relation` to the `users` collection; you can also enable the "Cascade delete" option)
- `like` (`bool` field; this is not required as you can also treat a missing record as "unlike")

You can also enable the "Cascade delete" for the above 2 relation fields so that when a post or user is deleted to automatically delete the related `postLikes` record(s).

To ensure that a user can vote only once, you can add a combined unique constraint for the `post` and `user` field.

To get the likes for a post, you can use [`expand` with indirect relation](https://pocketbase.io/docs/expanding-relations/#indirect-expand), something like:

```js
const post = await pb.collection("posts").getOne("POST_ID", {
    expand: "postLikes(post)"
})
```

Alternatively, if the "reactions" are just like/unlike, you can also create a helper [View collection](https://pocketbase.io/docs/collections/#view-collection) with a sum/count aggregation:

```sql
SELECT
    posts.id
    -- sum or just count(*) if you are storing only the likes
    sum(postLikes.like) as totalLikes
FROM posts
INNER JOIN postLikes on postLikes.post = posts.id
GROUP BY posts.id
```

**Reply by AaronHessler** [reply]
But how would I add a relation to my relation field? Is that even possible in the current version of pocketbase? I want to have multiple posts and I can't create a collection every single one.

![image](https://user-images.githubusercontent.com/122982679/235370801-4d5b13b5-722d-4404-9290-fbe984fa425f.png)
I just want to add a user like I can in this user interface.

**Reply by ganigeorgiev** [reply]
> But how would I add a relation to my relation field? Is that even possible in the current version of pocketbase? I want to have multiple posts and I can't create a collection every single one.

I'm sorry, I don't understand what do you mean. The suggested `postLikes` collection will store the likes for **all** posts.

**Reply by AaronHessler** [reply]
I think Im just going to make an end to end relation like you said! Thank you very very much!
